services/web/server/tests/unit/with_dbs/03/wallets/test_wallets.py

```diff
@@ -209,3 +212,24 @@ async def test_wallets_events_auto_add_default_wallet_on_user_confirmation(
     assert wallet.description == _WALLET_DESCRIPTION_TEMPLATE.format(user_name)
     assert mock_rut_sum_total_available_credits_in_the_wallet.called
     assert mock_add_credits_to_wallet.called == product.is_payment_enabled
+
+    # Test whether default wallet was set in user preferences
+    url = client.app.router["get_default_wallet"].url_for()
+    resp = await client.get(f"{url}")
+    data, _ = await assert_status(resp, web.HTTPOk)
+    assert data
+    wallet = WalletGetWithAvailableCredits(**data)
+    assert wallet.available_credits > Decimal(0)
+
+
+@pytest.mark.parametrize("user_role,expected", [(UserRole.USER, web.HTTPOk)])
+async def test_get_default_wallet_not_found(
+    client: TestClient,
+    logged_user: UserInfoDict,
+    expected: type[web.HTTPException],
+    wallets_clean_db: AsyncIterator[None],
+    mock_rut_sum_total_available_credits_in_the_wallet: mock.Mock,
+):
+    url = client.app.router["get_default_wallet"].url_for()
+    resp = await client.get(f"{url}")
+    await assert_status(resp, web.HTTPNotFound)
```

Comment:
IMO not found should only be used in collections, not in singletons like this. Now I would suggest the way to go as I mentioned above, i.e. `GET /wallets/default`. The client can then use exactly the same mechanisms to get and handle errors for one or multiple wallets